libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts

```diff
@@ -0,0 +1,130 @@
+// (C) 2022 GoodData Corporation
+
+import { createSelector } from "@reduxjs/toolkit";
+import {
+    DashboardState,
+    selectCanManageACL,
+    selectCanManageAnalyticalDashboard,
+    selectCanManageWorkspace,
+    selectDashboardLockStatus,
+    selectDashboardRef,
+    selectIsExport,
+    selectIsLayoutEmpty,
+    selectIsReadOnly,
+    selectListedDashboardsMap,
+} from "../../../../model";
+import {
+    selectDashboardEditModeDevRollout,
+    selectEnableAnalyticalDashboardPermissions,
+} from "../../../../model/store/config/configSelectors";
+import { selectIsDashboardLoading } from "../../../../model/store/loading/loadingSelectors";
+import {
+    selectDashboardShareStatusPrivate,
+    selectIsDashboardDirty,
+    selectIsNewDashboard,
+} from "../../../../model/store/meta/metaSelectors";
+import {
+    selectSupportsAccessControlCapability,
+    selectSupportsKpiWidgetCapability,
+} from "../../../../model/store/backendCapabilities/backendCapabilitiesSelectors";
+import { selectIsInEditMode } from "../../../../model/store/ui/uiSelectors";
+
+/**
+ * @internal
+ */
+export function selectCanEnterEditMode(state: DashboardState) {
```

Comment:
Please use `createSelector`, and provide rest of the selectors as a parameters, so the selector can be cached and not recalculated on every state change.

Reply:
I have used createSelector almost everywhere, but here, I prefer classical function because of early return - when dashboard is in readonly mode or rollout flag is disabled, we don't need to call another selectors, but maybe it is microoptimization....

Reply:
All the selectors inside are pointing to things, that are not changing (or not very often), so I think, that it will be more optimized with use of `createSelector` than by early return 🙂